src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,286 @@
+}
+
+
+export default class UserFileGrid extends React.Component<UserFileGridProps, UserFileGridState> {
+  modalTitle='Trash Submission?' 
+  modalCopy=<><p>This submission is currently incomplete and has not been submitted. If you trash this submission, you won't be able to recover the data.</p>
+  <p>Are you sure you want to trash this submission?' </p></>
+  
+  
+  constructor(props: UserFileGridProps) {
+    super(props)
+    this.state = {
+      isLoading: true,
+      token: this.props.token,
+      fileList: []
+    }
+  }
+
+  componentDidMount(): Promise<any> {
+    // Note:  All portals should do this once on the initial app load.
+    // This looks for the session token cookie (HttpOnly, unable to directly access), and initialize the session if it does exists.
+    SynapseClient.detectSSOCode()
+    return SynapseClient.getSessionTokenFromCookie()
+      .then((sessionToken: any) => { return this.refresh(sessionToken) })
+      .catch((error: any) => {
+        console.error(error)
+      })
+  }
+
+
+  refresh = (token?: string): Promise<any> => {
+    this.setState({ token })
+    if (token) {
+      return this.getUserFileListing(token!, this.props.parentContainerId).catch((error) => {
+        this.onError(error)
+      })
+
+    } else {
+      return Promise.reject()
+    }
+  }
+
+
+
+  private getUserFileListing = async (token: string, parentContainerId: string): Promise<UserData> => {
+    this.setState({ isLoading: true });
+    const result: UserData = {
+      fileList: [],
+      existingData: {},
+      dataFolderId: ''
+    }
+
+    // get user profile
+    const userprofile = await SynapseClient.getUserProfile(token)
+
+    //find directory named with profileId
+    const entityLookupRequest: EntityLookupRequest = {
+      entityName: userprofile.ownerId,
+      parentId: parentContainerId
+    }
+    try {
+      const entity = await SynapseClient.lookupChildEntity(entityLookupRequest, token)
+
+      //if found - that's the folder id
+      result.dataFolderId = entity.id
+    }
+
+    catch (error) {
+      if (error.statusCode === 404) {
+        // if doesn't exist - create
+        const entityId = await this.createDataFolder(userprofile, token)
+        if (!entityId) {
+          throw ('no folder')
+        }
+        result.dataFolderId = entityId;
+      } else {
+        this.onError(error)
+        return Promise.reject(error);
+      }
+    }
+    finally {
+      // get the file listing
+      result.fileList = await this.getFileListingInFolder(token, result.dataFolderId)
+      this.setState({
+        userprofile,
+        dataFolderId: result.dataFolderId,
+        fileList: result.fileList || [],
+        isLoading: false
+      })
+      return result;
+    }
+  }
+  onError = (args: any) => {
+    console.log(args)
+  }
+
+
+  // creates data folder for user
+  createDataFolder = async (userprofile: UserProfile, token: string): Promise<string | undefined> => {
```

Comment:
TODO: create folder and create ACL (see https://sagebionetworks.jira.com/browse/PLFM-5692) allowing only the current user access (private folder containing their form data).

Reply:
Yes. I will add it to the TODO: in the top of UserFileGrid.tsx